db/migrate/20181219043340_add_fields_to_node.rb

```diff
@@ -0,0 +1,35 @@
+class AddFieldsToNode < ActiveRecord::Migration[5.2]
+  def up
+    add_column :node, :latitude, :decimal, :precision => 20, :scale => 17
+    add_column :node, :longitude, :decimal, :precision => 20, :scale => 17
+    add_column :node, :precision, :integer
+
+    sql = "SELECT `node`.`nid`,`term_data`.`name` FROM `node` LEFT OUTER JOIN `community_tags` ON `community_tags`.`nid` = `node`.`nid` LEFT OUTER JOIN `term_data` ON `term_data`.`tid` = `community_tags`.`tid` WHERE (term_data.name LIKE 'lat:%' or term_data.name LIKE 'lon:%')"
+    tags_array = ActiveRecord::Base.connection.exec_query(sql)
+    tags_array.each do |tag|
+      node_id = tag["nid"]
+      if tag["name"].include? 'lon:'
+        long = tag["name"].gsub("lon:","").to_f
+        update "UPDATE `node` SET `precision` = '" + decimals(long).to_s + "' WHERE nid = '" + node_id.to_s + "'"
+        update "UPDATE `node` SET `longitude` = '" + long.to_s + "' WHERE nid = '" + node_id.to_s + "'"
+      else
+        lati = tag["name"].gsub("lat:","")
+        update "UPDATE `node` SET `latitude` = '" + lati + "' WHERE nid = '" + node_id.to_s + "'"
```

Comment:
Hi @milaaraujo it looks like this didn't run correctly on production for some reason -- odd that it didn't fail on stable, but i'm looking into it now. I think in general we may want to stick to non-raw SQL in the future if possible as there may be more safeguards... but i don't really know that this is the issue right now:

```
Mysql2::Error: Incorrect decimal value: 'false' for column 'latitude' at row 1
ActiveRecord::StatementInvalid: Mysql2::Error: Incorrect decimal value: 'false' for column 'latitude' at row 1: UPDATE `node` SET `latitude` = 'false' WHERE nid = '2120'
```

I wonder if some of the lat/lon tags are not supplying numbers!!??

Reply:
Aha! I think this tag exists, though not sure where: https://publiclab.org/tag/lat:false

OK, i'll try to manually remove it and re-run the migration. If that doesn't work, we may want to amend the migration to only run if the latitude/longitude values are actually numbers...

Reply:
OK, deleted `lat:false` as a tag. I tried scanning on the console for any other non-numerical ones and wasn't able to find any. Hopefully it'll run properly this time, restarting! Sorry for the scare! This is publishing to production now, exciting!!!

Reply:
Did it work?